controllers/managedocs_controller.go

```diff
@@ -255,6 +265,95 @@ func (r *ManagedOCSReconciler) SetupWithManager(mgr ctrl.Manager, ctrlOptions *c
 		},
 	)
 
+	var options client.Options
+	var err error
+	var unrestrictedClient client.Client
+
+	options.Scheme = runtime.NewScheme()
+	utilruntime.Must(apiextenstionsv1.AddToScheme(options.Scheme))
+
+	unrestrictedClient, err = client.New(config.GetConfigOrDie(), options)
+	if err != nil {
+		r.Log.Error(err, "error creating client")
+		os.Exit(1)
+	}
+
+	egreeNetworkPolicyCRD := &unstructured.Unstructured{}
+	egreeNetworkPolicyCRD.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "apiextensions.k8s.io",
+		Kind:    "CustomResourceDefinition",
+		Version: "v1",
+	})
+	ctx := context.TODO()
+	defer ctx.Done()
+	if err = unrestrictedClient.Get(context.TODO(), client.ObjectKey{Name: "egressnetworkpolicies.network.openshift.io", Namespace: ""}, egreeNetworkPolicyCRD); err != nil {
+		if strings.Contains(err.Error(), "could not find the requested resource") {
```

Comment:
- it's not a good idea to have this code duplicated
- `NewControllerManagedBy` returns a builder and you can keep on calling functions off of it, why not use that pattern?
- Add the common resources and based on the error decide b/n `EgressNetworkPolicy` and `EgressFirewall`

Reply:
done

Reply:
ack